src/mod/common/nl/nl_common.c

```diff
@@ -14,7 +14,7 @@ char *get_iname(struct genl_info *info)
 
 struct joolnlhdr *get_jool_hdr(struct genl_info *info)
 {
-	return info->userhdr;
+	return (struct joolnlhdr *)(info->genlhdr + GENL_HDRLEN);
```

Comment:
`struct genlmsghdr` is a [four byte structure](https://elixir.bootlin.com/linux/v5.15/source/include/uapi/linux/genetlink.h#L13) (excluding unlikely padding and slop), which means if you add a number to a genlmsghdr pointer, you will actually sum four times the indicated amount.

In other words, you're returning the equivalent of

```
((u8 *)info->genlhdr) + 4 * GENL_HDRLEN
```

When you actually want to return

```
((u8 *)info->genlhdr) + GENL_HDRLEN
```

Try compiling the following userspace program to see the difference:

```
#include <stdio.h>
#include <linux/genetlink.h>

int main(int argc, char **argv)
{
	struct genlmsghdr base;
	struct genlmsghdr *ptr;

	ptr = &base;

	printf("ptr              : %p\n", ptr);
	/* Prints the location right next to base. */
	printf("ptr + 1          : %p\n", ptr + 1);
	/* Prints a location far to base's right */
	printf("ptr + GENL_HDRLEN: %p\n", ptr + GENL_HDRLEN);

	return 0;
}
```

(Dump that into a file called pointer-test.c, then run something like `gcc pointer-test.c && ./a.out`)